packages/swapkit/helpers/src/helpers/memo.ts

```diff
@@ -91,3 +128,277 @@ export const getMemoFor = <T extends MemoType>(memoType: T, options: MemoOptions
       return "";
   }
 };
+
+export async function parseMemo(memo: string) {
+  const memoParts = memo.split(":");
+  const [actionPartUnparsed] = memoParts;
+
+  if (!actionPartUnparsed) throw new Error("No memo action found");
+
+  const memoAction = parseMemoAction(actionPartUnparsed);
+
+  switch (memoAction) {
+    case MemoType.BOND:
+      return parseBondMemo(memo);
+    case MemoType.UNBOND:
+      return parseUnbondMemo(memo);
+    case MemoType.LEAVE:
+      return parseLeaveMemo(memo);
+    case MemoType.CLOSE_LOAN:
+      return parseCloseLoanMemo(memo);
+    case MemoType.OPEN_LOAN:
+      return parseOpenLoanMemo(memo);
+    case MemoType.WITHDRAW:
+      return parseWithdrawMemo(memo);
+    case MemoType.DEPOSIT:
+      return parseDepositMemo(memo);
+    case MemoType.SWAP:
+      return await parseSwapMemo(memo);
+    default:
+      throw new Error("Unknown memo action");
+  }
+}
+
+// parseMemoAction('=')
+// parseMemoAction('SWAP')
+// MemoType.SWAP
+const parseMemoAction = (action: string): MemoType | undefined => {
```

Comment:
If that's comment - why it's not required by type?
```suggestion
// parseMemoAction('=')
// parseMemoAction('SWAP')
// MemoType.SWAP
const parseMemoAction = (action: 'SWAP' | '='): MemoType | undefined => {
```